hikari/impl/interaction_server.py

```diff
@@ -0,0 +1,558 @@
+
+            self._verify = ed25519.build_ed25519_verifier(application.public_key)
+            return self._verify
+
+    async def aiohttp_hook(self, request: aiohttp.web.Request) -> aiohttp.web.Response:
+        """Handle an AIOHTTP interaction request.
+
+        This method handles aiohttp specific detail before calling
+        `InteractionServer.on_interaction` with the data extracted from the
+        request if it can and handles building an aiohttp response.
+
+        Parameters
+        ----------
+        request : aiohttp.web.Request
+            The received request.
+
+        Returns
+        -------
+        aiohtttp.web.Response
+            The aiohttp response.
+        """
+        if request.content_type.lower() != _JSON_CONTENT_TYPE:
+            return aiohttp.web.Response(
+                status=_UNSUPPORTED_MEDIA_TYPE_STATUS, body="Unsupported Media Type", content_type=_TEXT_CONTENT_TYPE
+            )
+
+        try:
+            signature_header = bytes.fromhex(request.headers[_X_SIGNATURE_ED25519_HEADER])
+            timestamp_header = request.headers[_X_SIGNATURE_TIMESTAMP_HEADER].encode()
+
+        except (KeyError, ValueError):
+            user_agent = request.headers.get(_USER_AGENT_KEY, "NONE")
+            _LOGGER.info("Received a request with a missing or invalid ed25519 header (UA %r)", user_agent)
+            return aiohttp.web.Response(
+                status=_BAD_REQUEST_STATUS,
+                body="Missing or invalid required request signature header(s)",
+                content_type=_TEXT_CONTENT_TYPE,
+            )
+
+        try:
+            body = await request.read()
+
+        except aiohttp.web.HTTPRequestEntityTooLarge:
+            return aiohttp.web.Response(
+                status=_PAYLOAD_TOO_LARGE_STATUS, body="Payload too large", content_type=_TEXT_CONTENT_TYPE
+            )
+
+        if not body:
+            user_agent = request.headers.get(_USER_AGENT_KEY, "NONE")
+            _LOGGER.info("Received a body-less request (UA %r)", user_agent)
+            return aiohttp.web.Response(
+                status=_BAD_REQUEST_STATUS, body="POST request must have a body", content_type=_TEXT_CONTENT_TYPE
+            )
+
+        response = await self.on_interaction(body=body, signature=signature_header, timestamp=timestamp_header)
+        return aiohttp.web.Response(status=response.status_code, headers=response.headers, body=response.payload)
+
+    async def close(self) -> None:
+        if not self._runner:
+            raise errors.ComponentStateConflictError("Cannot close an inactive interaction server")
+
+        runner = self._runner
+        future = self._future
+        self._runner = None
+        await runner.shutdown()
+        await runner.cleanup()
+        future.set_result(None)
+
+    async def join(self) -> None:
+        if not self._runner:
+            raise errors.ComponentStateConflictError("Cannot wait for an inactive interaction server to join")
```

Comment:
This would infer that "joining" it actively does anything as opposed to being a passive operation

Reply:
> 
> 
> This would infer that "joining" it actively does anything as opposed to being a passive operation

Oh, true. I was thinking in the bot context. My bad